public/src/components/portfolio/Portfolio.tsx

```diff
@@ -37,12 +37,25 @@ export interface PortfolioProps {
 }
 
 interface PortfolioState {
+    userReservedStocks: number
 }
 
 
 export class Portfolio extends React.Component<PortfolioProps, PortfolioState> {
     constructor(props: PortfolioProps) {
         super(props);
+        var stocksReservedMap = this.props.stocksReservedMap;
+        var userReservedStocks = 0;
+        console.log("hiii");
+        console.log(this.props.stocksReservedMap);
```

Comment:
You do not need to calculate Reserved stocks here.. Pass them as props from Main.tsx